### PR TITLE
Minor build optimizations for libraries

### PR DESCRIPTION
**Pull request by ryanking13**
Minor build optimizations for libraries by:

- Not building executables and test if possible
- Using `--disable-dependency-tracking` flags for libraries that use autoconf

### CODE REVIEW

**Review by hoodmane — APPROVED** [review]
Looks good to me.

File: packages/liblzma/meta.yaml [hunk 1]
```diff
@@ -19,6 +19,7 @@ build:
         --disable-scripts \
         --disable-doc \
         --enable-shared=no \
+        --disable-dependency-tracking \
```
Comment (rth) [inline packages/liblzma/meta.yaml:22]: I think it's disabled by default already? https://www.gnu.org/software/automake/manual/html_node/Dependency-Tracking.html

Comment (ryanking13) [reply packages/liblzma/meta.yaml:22]: I think it is not disabled by default. I can observe the log in main:

```
checking dependency style of /root/repo/emsdk/emsdk/upstream/emscripten/emcc... gcc3
```

which is changed in this PR:

```
checking dependency style of /root/repo/emsdk/emsdk/upstream/emscripten/emcc... none
```

Comment (rth) [reply packages/liblzma/meta.yaml:22]: OK, thanks for checking

**Pull request merged**